src/hypothesis/strategies.py

```diff
@@ -1625,10 +1625,16 @@ def check_valid_bound(value, name):
     """
     if value is None or isinstance(value, integer_types + (Rational,)):
         return
-    if math.isnan(value):
-        raise InvalidArgument(u'Invalid end point %s %r' % (value, name))
+    is_nan = False
+    try:
+        is_nan = math.isnan(value)
+    except TypeError:
+        pass
```

Comment:
Should `value` really be considered a valid endpoint in the case?  What's an example of this?  (`Decimal('snan')` would be my counterexample, and raise a ValueError to boot)